components/region_cache_memory_engine/src/background.rs

```diff
@@ -675,4 +799,78 @@ pub mod tests {
         let key = encoding_for_filter(key.as_encoded(), TimeStamp::new(commit_ts4));
         assert!(write.get(&key).is_some());
     }
+
+    #[test]
+    fn test_background_worker_load() {
+        let mut engine = RangeCacheMemoryEngine::new(Arc::default(), Duration::from_secs(1000));
+        let path = Builder::new().prefix("test_load").tempdir().unwrap();
+        let path_str = path.path().to_str().unwrap();
+        let rocks_engine = new_engine(path_str, DATA_CFS).unwrap();
+        engine.set_disk_engine(rocks_engine.clone());
+
+        for i in 10..20 {
+            let key = construct_key(i, 1);
+            let key = data_key(&key);
+            let value = construct_value(i, i);
+            rocks_engine
+                .put_cf(CF_DEFAULT, &key, value.as_bytes())
+                .unwrap();
+            rocks_engine
+                .put_cf(CF_WRITE, &key, value.as_bytes())
+                .unwrap();
+        }
+
+        let k = format!("zk{:08}", 15).into_bytes();
+        let r1 = CacheRange::new(DATA_MIN_KEY.to_vec(), k.clone());
+        let r2 = CacheRange::new(k, DATA_MAX_KEY.to_vec());
+        {
+            let mut core = engine.core.write().unwrap();
+            core.mut_range_manager().pending_ranges.push(r1.clone());
+            core.mut_range_manager().pending_ranges.push(r2.clone());
+        }
+        engine.handle_pending_load();
+
+        // concurrent write to rocksdb, but the key will not be loaded in the memory
+        // engine
+        let key = construct_key(20, 1);
+        let key20 = data_key(&key);
+        let value = construct_value(20, 20);
+        rocks_engine
+            .put_cf(CF_DEFAULT, &key20, value.as_bytes())
+            .unwrap();
+        rocks_engine
+            .put_cf(CF_WRITE, &key20, value.as_bytes())
+            .unwrap();
+
+        let (write, default) = {
+            let core = engine.core().write().unwrap();
+            let skiplist_engine = core.engine();
+            (
+                skiplist_engine.cf_handle(CF_WRITE),
+                skiplist_engine.cf_handle(CF_DEFAULT),
+            )
+        };
+
+        // wait for background load
+        std::thread::sleep(Duration::from_secs(1));
```

Comment:
if you support passing a callback(`Box<dyn FnOnce(Vec<CacheRange>) + Send>`) to pending_load (and subsequently to `LoadTask`), you can avoid a sleep. e.g.,

```rust
let (cb, fut) = paired_future_callback::<Vec<CacheRange>>::();
engine.handle_pending_load(cb); 
...
let _ = block_on(f1);
```